clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -345,6 +345,18 @@ RValue CIRGenFunction::emitRotate(const CallExpr *E, bool IsRotateRight) {
   return RValue::get(r);
 }
 
+static bool isMemBuiltinOutOfBoundPossible(const CallExpr *expr,
+                                           clang::CIRGen::CIRGenModule &cgm,
```

Comment:
Maybe we can take the ASTContext as a parameter instead of the cgm, since that's all we're using? That'll let us potentially share this helper with CodeGen in the future.

Reply:
yeah, it's a good idea, will change it. Also, now looking at it. I'm also thinking maybe I should pass argument expr of dstsize and size as parameters, instead of passing CallExpr so we have to do  E->getArg(2) which assumes that size and dst size are 3rd and 4th argument.

Reply:
Yup, that's a good idea.